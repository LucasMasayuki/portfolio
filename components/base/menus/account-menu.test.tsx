import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import AccountMenu from './account-menu';

describe.concurrent('Account Menu component', () => {
  describe('when render the first time', () => {
    it('should not open menu', () => {
      const { container } = render(<AccountMenu />);

      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when clicked in photo avatar', () => {
    it('should open menu', () => {
      const { container } = render(<AccountMenu />);

      const photoAvatarComponent = container.querySelector('button');

      photoAvatarComponent?.click();

      expect(container.innerHTML).toMatchSnapshot();
    });
  });
});
