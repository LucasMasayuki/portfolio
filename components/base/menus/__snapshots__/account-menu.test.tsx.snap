// Vitest Snapshot v1

exports[`Account Menu component > when clicked in photo avatar > should open menu 1`] = `"<div class=\\"MuiBox-root css-5nwj3y\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Conta\\" data-mui-internal-clone-element=\\"true\\"><div class=\\"MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-6u9n7p-MuiAvatar-root\\">T</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div>"`;

exports[`Account Menu component > when render the first time > should not open menu 1`] = `"<div class=\\"MuiBox-root css-5nwj3y\\"><button class=\\"MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Conta\\" data-mui-internal-clone-element=\\"true\\"><div class=\\"MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-6u9n7p-MuiAvatar-root\\">T</div><span class=\\"MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root\\"></span></button></div>"`;
